### PR TITLE
Pytest

### PR DESCRIPTION
**Pull request by AHReccese**
#### What does this implement/fix? Explain your changes.
* Make all tests runnable by pytest
* update test.yml test scripts

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -30,10 +30,10 @@ jobs:
           python setup.py install 
       - name: Test Linear Models
         run: |
-          python tests/test_linear_models/test_all.py
+          python tests/test_linear_models/test_linear_models.py
       - name: Test Pymilo Exceptions
         run: |
-          python tests/test_exceptions/test_all.py
+          python tests/test_exceptions/test_exceptions.py
```
Comment (sadrasabouri) [inline .github/workflows/test.yml:36]: Don't you want to add Pytest to the `test.yml`? (like we do [here](https://github.com/sepandhaghighi/samila/blob/master/.github/workflows/test.yml#L40))

File: tests/test_linear_models/test_linear_models.py [hunk 2]
```diff
@@ -125,15 +125,13 @@ def reset_exported_models_directory(self):
...
-            category_all_test_pass = True
             for model in self.LINEAR_MODELS[category]:
                 if isinstance(model, tuple):
                     func, model_name = model
                     if func == None:
                         print("Model: " + model_name + " is not supported in this python version.")
                         continue
-                category_all_test_pass = category_all_test_pass and model()
-                self.assertTrue(category_all_test_pass)
+                model()
```
Comment (sadrasabouri) [inline tests/test_linear_models/test_linear_models.py:134]: Can you explain why this change is required?

**Review by sadrasabouri — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (AHReccese) [reply .github/workflows/test.yml:36]: you are right, sure.

File: tests/test_linear_models/test_linear_models.py (hunk 2 above)
Comment (AHReccese) [reply tests/test_linear_models/test_linear_models.py:134]: Pytest-associated test functions are warned to return values and they will not be supported in the next versions of Pytest, so as you can see the `model` which is the test function of the associated model, doesn't return anything and it just has an `assertion`, so I removed `category_all_test_pass ` variable.

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Thanks for your efforts 🔥 
Please add this section to `test.yml`:
```
- name: Upload coverage to Codecov
      uses: codecov/codecov-action@v3
      with:
          fail_ci_if_error: false
      if: matrix.python-version == env.TEST_PYTHON_VERSION && matrix.os == env.TEST_OS
```

like [here](https://github.com/sepandhaghighi/pyrgg/blob/6a5d63f6e9398cdc83d3350f5299af53049bc92a/.github/workflows/test.yml#L51)

**Review by sadrasabouri — APPROVED** [review]
LGTM :rocket:

File: dev-requirements.txt [hunk 3]
```diff
@@ -1,4 +1,8 @@
 setuptools>=40.8.0
 vulture>=1.0
 bandit>=1.5.1
-pydocstyle>=3.0.0
\ No newline at end of file
+pydocstyle>=3.0.0
+codecov>=2.0.15
```
Comment (sepandhaghighi) [inline dev-requirements.txt:5]: @AHReccese Please remove `codecov` from here.
This package is deprecated and it has caused many problems in the past

File: dev-requirements.txt [hunk 4]
```diff
@@ -1,4 +1,8 @@
...
+coverage>=4.1
```
Comment (sepandhaghighi) [inline dev-requirements.txt:6]: Also, remove `coverage`.
It's not necessary here

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@AHReccese Take a look at my comments 🔥

**Review by sepandhaghighi — APPROVED** [review]
